expression/builtin_cast_vec.go

```diff
@@ -690,11 +690,42 @@ func (b *builtinCastDurationAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, res
 }
 
 func (b *builtinCastIntAsDecimalSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastIntAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	isUnsigned := mysql.HasUnsignedFlag(b.tp.Flag) || mysql.HasUnsignedFlag(b.args[0].GetType().Flag)
+	nums := buf.Int64s()
+	result.ResizeDecimal(n, false)
+	result.MergeNulls(buf)
+	decs := result.Decimals()
+	sc := b.ctx.GetSessionVars().StmtCtx
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if !isUnsigned {
+			decs[i].FromInt(nums[i])
+		} else if b.inUnion && nums[i] < 0 {
+			decs[i].FromUint(0)
+		} else {
+			decs[i].FromUint(uint64(nums[i]))
+		}
+		if _, err := types.ProduceDecWithSpecifiedTp(&decs[i], b.tp, sc); err != nil {
```

Comment:
It seems we should use a variable to store its result and copy the result back into `decs[i]` since this function may create a new `MyDecimal`.

Reply:
It seems you are right.
I copied this usage from `func (b *builtinCastTimeAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Column)`, should I correct it together?

Reply:
Yes, please fix it together in this PR.

Reply:
Done.